control/init.sh

```diff
@@ -35,8 +34,15 @@ echo "Downloading dependencies from galaxy"
 export ANSIBLE_ROLES_PATH=~/.ansible/roles
 ansible-galaxy install -r ~/commcarehq-ansible/ansible/requirements.yml
 
+pip install -r ~/commcarehq-ansible/ansible/requirements.txt
```

Comment:
this was purposely left out in the past because it makes the init script take _much_ longer to run, and many people have it in their profile, which can make starting a new shell take an inconvenient amount of time.

Reply:
Yeah, I considered this. It definitely puts a bunch more useless output on the screen (which of course we could clean up), but the total start time from hitting enter on your `ssh` command to receiving a usable prompt is something like 5 seconds in steady state. (I just timed it.) Of course, you're right that it will be longer than that if you have an install you have to do. That's rarely the case, but when it _is_ the case, my guess is that auto-updating will save you more time and energy debugging your system after a minor upgrade than you'll lose waiting for it.

My primary goal here is to make the context from which we run cloud maintenance commands as self-managing as possible so that we don't all have to be experts in the nitty-gritty details of our setup.

What do you think?

Reply:
My concern is that sometimes the control machine is also used for other things (db0 on enikshay), and I don't necessarily want to wait for something I'm not even doing when I log into that machine. I guess I will just remove the init script from my profile and run it manually when I want to use ansible.

Another idea would be to remove the init script entirely and do those things as part of `commcare-cloud`.

Reply:
Yeah, I see your point. To make it at least a little faster, I opened https://github.com/dimagi/commcarehq-ansible/pull/1050, but agree this is ripe for rethinking soon